lib/index.ts

```diff
@@ -77,7 +77,9 @@ export async function fetchBeacon(client: ChainClient, roundNumber?: number): Pr
         beacon = await client.get(roundNumber)
     }
 
-    return validatedBeacon(client, beacon)
+    const expectedRound = roundAt(Date.now(), await client.chain().info())
```

Comment:
This should take into account the case where roundNumber is provided.
```suggestion
    const expectedRound = roundNumber || roundAt(Date.now(), await client.chain().info())
```

Reply:
oops yeah I refactored this out by accident hehe